petastorm/tests/test_parquet_reader.py

```diff
@@ -86,3 +85,35 @@ def test_string_partition(reader_factory, tmpdir, partition_by):
         row_ids_batched = [row.id for row in reader]
     actual_row_ids = list(itertools.chain(*row_ids_batched))
     assert len(data) == len(actual_row_ids)
+
+
+@pytest.mark.parametrize('reader_factory', _D)
+def test_column_subset(scalar_dataset, reader_factory):
+    """Request subset of columns from reader, confirm that receive those columns and only those columns."""
+
+    # Create field subset, by picking even-numbered fields from the available fields, counting from 0.
+    all_fields = sorted(scalar_dataset.data[0].keys())
+    requested_fields = []
+    for n in range(0, len(all_fields), 2):
+        requested_fields.append(all_fields[n])
+
+    with reader_factory(scalar_dataset.url, schema_fields=requested_fields) as reader:
+        sample = next(reader)
+        assert sorted(sample._asdict().keys()) == requested_fields
+
+
+@pytest.mark.parametrize('reader_factory', _D)
+def test_invalid_column_name(scalar_dataset, reader_factory):
+    """Request a column that doesn't exist, confirm that get exception."""
+    # Grab first field from expected dataset, append random characters to it to get an invalid field name.
+    all_fields = list(scalar_dataset.data[0].keys())
+    bad_field = all_fields[0]
+    while bad_field in all_fields:
+        bad_field += "VR46"
+    requested_fields = [bad_field]
+
+    with pytest.raises(StopIteration):
```

Comment:
`StopIteration` is not really an error, but just an indicator of the end of the data. Would `ValueError` be more appropriate here?

Reply:
When requesting only invalid fields, it seems that DummyPool returns an EmptyResultError that ArrowReaderWorkerResultsQueueReader turns into a StopIteration exception. Is there a better way to test this, or should this type of request return something different?